run_configs/lfric/lfric_common.py

```diff
@@ -21,79 +23,113 @@ def check_available(self):
-                                           '-directory', str(config_dir),
+    rose_picker.run(additional_parameters=[rose_meta_conf,
+                                           '-directory', config_dir,
                                            '-include_dirs', lfric_source],
                     env=env)
+    rose_meta = config_dir / 'rose-meta.json'
 
-    # "build_config_loaders"
+    # build_config_loaders
+    # --------------------
     # builds a bunch of f90s from the json
     logger.info('GenerateNamelist')
     gen_namelist = Script(gen_namelist_tool)
-    gen_namelist.run(additional_parameters=['-verbose',
-                                            str(config_dir / 'rose-meta.json'),
-                                            '-directory', str(config_dir)])
+    gen_namelist.run(additional_parameters=['-verbose', rose_meta,
+                                            '-directory', config_dir],
+                     cwd=config_dir)
 
     # create configuration_mod.f90 in source root
+    # -------------------------------------------
     logger.info('GenerateLoader')
+    names = [name.strip() for name in
+             open(config_dir / 'config_namelists.txt').readlines()]
+    configuration_mod_fpath = config_dir / 'configuration_mod.f90'
     gen_loader = Script(gen_loader_tool)
-    names = [name.strip() for name in open(config_dir / 'config_namelists.txt').readlines()]
-    configuration_mod_fpath = config.source_root / 'configuration_mod.f90'
     gen_loader.run(additional_parameters=[configuration_mod_fpath,
                                           *names])
 
     # create feign_config_mod.f90 in source root
+    # ------------------------------------------
     logger.info('GenerateFeigns')
-    feign_config = Script(gen_feigns_tool)
-    feign_config_mod_fpath = config.source_root / 'feign_config_mod.f90'
-    feign_config.run(additional_parameters=[str(config_dir / 'rose-meta.json'),
-                                            '-output', feign_config_mod_fpath])
+    feign_config_mod_fpath = config_dir / 'feign_config_mod.f90'
+    gft = Script(gen_feigns_tool)
+    gft.run(additional_parameters=[rose_meta,
+                                   '-output', feign_config_mod_fpath])
 
-    # put the generated source into an artefact
-    # todo: we shouldn't need to do this, should we?
-    #       it's just going to be found in the source folder with everything else.
-    config._artefact_store['configurator_output'] = [
-        configuration_mod_fpath,
-        feign_config_mod_fpath
-    ]
+    find_source_files(config, source_root=config_dir)
 
 
+# ============================================================================
 @step
 def fparser_workaround_stop_concatenation(config):
     """
-    fparser can't handle string concat in a stop statement. This step is a workaround.
+    fparser can't handle string concat in a stop statement. This step is
+    a workaround.
```

Comment:
I wonder if this is still needed? If you are feeling diligent you might find the fParser issue number and include it in the comment.

Reply:
The issue number (330) is just in the next line. And the ticket is still open. I've labelled it for the next release, tbh, it shouldn't be that much work, I will try to get it done.